tests/test_file_uploader.py

```diff
@@ -0,0 +1,323 @@
+    mocked_post = mocker.patch.object(requests, "post", autospec=True)
+
+    expected_open_file = mocked_open.return_value.__enter__()
+    test_file_name = "fname"
+    test_file_path = "/tmp/fname"
+    test_url = "website.com"
+    test_data = {"key": "val"}
+    test_upload_details = {"presigned_params": {"url": test_url, "fields": test_data}}
+
+    upload_file_to_s3(test_file_path, test_file_name, test_upload_details)
+
+    mocked_open.assert_called_once_with(test_file_path, "rb")
+    mocked_post.assert_called_once_with(
+        test_url, data=test_data, files={"file": (test_file_name, expected_open_file)}
+    )
+
+
+def test_get_sdk_status__requests_and_returns_sdk_status_correctly(mocker):
+    mocked_post = mocker.patch.object(requests, "get", autospec=True)
+
+    expected_sdk_status = mocked_post.return_value.json()["status"]
+    test_access_token = "token"
+    test_upload_details = {"upload_id": "test_id"}
+
+    actual = get_sdk_status(test_access_token, test_upload_details)
+    mocked_post.assert_called_once_with(
+        "https://<TODO>.execute-api.us-east-1.amazonaws.com/prod-lambda-gw-stage/get_sdk_status?upload_id=test_id",
+        headers={"Authorization": f"Bearer {test_access_token}"},
+    )
+    assert actual == expected_sdk_status
+
+
+def test_create_zip_file__correctly_writes_h5_files_to_zipfile_at_designated_path(mocker):
+    mocked_zip_function = mocker.patch.object(zipfile, "ZipFile", autospec=True)
+    mocked_os_walk = mocker.patch.object(os, "walk", autospec=True)
+    spied_os_join = mocker.spy(os.path, "join")
+    mocked_os_walk.return_value = [
+        (
+            "/tmp/test_h5_files",
+            ("",),
+            (
+                "test_1.h5",
+                "test_2.h5",
+            ),
+        ),
+    ]
+
+    test_dir_path = "/tmp"
+    test_file_name = "test_h5_files"
+    test_zipped_path = f"{test_dir_path}/zipped_recordings/cid"
+
+    zipped_file_path = create_zip_file(test_dir_path, test_file_name, test_zipped_path)
+    mocked_zip_function.assert_called_once_with(f"{os.path.join(test_zipped_path, test_file_name)}.zip", "w")
+    assert len(spied_os_join.call_args_list) == 5
+    assert zipped_file_path == f"{os.path.join(test_zipped_path, test_file_name)}.zip"
+
+
+def test_create_zip_file__create_zip_file_should_not_be_called_with_previously_failed_zip_file(mocker):
+    mocked_create_zip_file = mocker.patch.object(file_uploader, "create_zip_file", autospec=True)
+    mocker.patch.object(file_uploader, "get_file_md5", autospec=True)
+    mocker.patch.object(file_uploader, "get_access_token", autospec=True)
+    mocker.patch.object(file_uploader, "get_upload_details", autospec=True)
+    mocker.patch.object(file_uploader, "upload_file_to_s3", autospec=True)
+    mocker.patch.object(file_uploader, "get_sdk_status", autospec=True, return_value="analysis complete")
+
+    test_file_name = "zip_file"
+    test_file_path = "/test"
+    test_zip_dir = "/test/zipped_recordings"
```

Comment:
This seems to work fine here, but sometimes you can run into issues when testing on different OS's when adding the separator character to a string directly since it's `/` on unix based OS's and `\` on windows. You can get around this issue by using `os.path.join("test", "zipped_recordings") which will insert the correct char depending on the OS

Reply:
Yes! i actually had that error and had to swap a lot of them.. idk why this one didn't error